envoy/network/socket.h

```diff
@@ -232,6 +248,11 @@ class ConnectionInfoProvider {
    * @return roundTripTime of the connection
    */
   virtual const absl::optional<std::chrono::milliseconds>& roundTripTime() const PURE;
+
+  /**
+   * @return the filter chain info provider backing this socket.
```

Comment:
When I see this in code it does feel that it's a bit weird in connection vs. stream (previous conversation notwithstanding). @ggreenway can you sanity check that what I suggested before is a net improvement here?

Reply:
I didn't get it. Because the filter chain info is config thing?

Reply:
You're referring to putting it into the `ConnectionInfoProvider` (or similar for filter chain)? Yes, I think that makes sense as a place for things that the stream inherits from the connection.